hedera-node/test-clients/src/main/java/com/hedera/services/bdd/suites/integration/LegacySystemAccountCleanupTest.java

```diff
@@ -0,0 +1,212 @@
+import static com.hedera.services.bdd.junit.hedera.subprocess.ProcessUtils.conditionFuture;
+import static com.hedera.services.bdd.junit.restart.RestartType.UPGRADE_BOUNDARY;
+import static com.hedera.services.bdd.junit.support.StreamFileAccess.STREAM_FILE_ACCESS;
+import static com.hedera.services.bdd.spec.HapiSpec.hapiTest;
+import static com.hedera.services.bdd.spec.transactions.TxnVerbs.cryptoTransfer;
+import static com.hedera.services.bdd.spec.transactions.crypto.HapiCryptoTransfer.tinyBarsFromTo;
+import static com.hedera.services.bdd.spec.utilops.UtilVerbs.blockingOrder;
+import static com.hedera.services.bdd.spec.utilops.UtilVerbs.withOpContext;
+import static com.hedera.services.bdd.suites.HapiSuite.FUNDING;
+import static com.hedera.services.bdd.suites.HapiSuite.GENESIS;
+import static com.hedera.services.bdd.suites.HapiSuite.ONE_BILLION_HBARS;
+import static com.hedera.services.bdd.suites.HapiSuite.ONE_HBAR;
+import static com.hederahashgraph.api.proto.java.HederaFunctionality.CryptoTransfer;
+import static java.util.Objects.requireNonNull;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNull;
+
+import com.hedera.hapi.node.base.AccountID;
+import com.hedera.hapi.node.base.HederaFunctionality;
+import com.hedera.hapi.node.state.entity.EntityCounts;
+import com.hedera.hapi.node.state.token.Account;
+import com.hedera.node.app.fixtures.state.FakeState;
+import com.hedera.node.app.hapi.utils.forensics.RecordStreamEntry;
+import com.hedera.node.app.ids.EntityIdService;
+import com.hedera.node.app.service.token.TokenService;
+import com.hedera.services.bdd.junit.TargetEmbeddedMode;
+import com.hedera.services.bdd.junit.hedera.NodeSelector;
+import com.hedera.services.bdd.junit.restart.RestartHapiTest;
+import com.hedera.services.bdd.junit.restart.SavedStateSpec;
+import com.hedera.services.bdd.spec.SpecOperation;
+import com.hederahashgraph.api.proto.java.AccountAmount;
+import com.swirlds.state.test.fixtures.MapWritableKVState;
+import com.swirlds.state.test.fixtures.MapWritableStates;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import java.io.File;
+import java.io.IOException;
+import java.io.UncheckedIOException;
+import java.nio.file.Path;
+import java.time.Instant;
+import java.util.HashSet;
+import java.util.Set;
+import java.util.SplittableRandom;
+import java.util.concurrent.TimeUnit;
+import java.util.function.LongFunction;
+import java.util.stream.IntStream;
+import java.util.stream.LongStream;
+import java.util.stream.Stream;
+import org.assertj.core.api.Assertions;
+import org.junit.jupiter.api.DynamicTest;
+import org.junit.jupiter.api.Order;
+import org.junit.jupiter.api.Tag;
+
+/**
+ * A test that "picks up" after a simulated restart with legacy system accounts in state, and verifies that they
+ * are cleaned up as expected by dispatching synthetic {@link HederaFunctionality#CRYPTO_DELETE} txs.
```

Comment:
Just a tweak, removing any reference to dispatching synthetic crypto deletes in the comment. Not a PR blocker

```suggestion
 * A test that "picks up" after a simulated restart with legacy system accounts in state, and verifies that they
 * are cleaned up as expected.
```